source/write-operations/replace-documents.txt

```diff
@@ -0,0 +1,217 @@
+datasets, see the :ref:`<java-rs-getting-started>` tutorial.
+
+Replace Operation
+-----------------
+
+You can perform a replace operation by using the ``replaceOne()``
+method on a ``MongoCollection`` instance. This method removes all fields except the ``_id`` field from the first
+document that matches the search criteria, then adds the fields and values
+you specify to the empty document.
+
+Required Parameters
+~~~~~~~~~~~~~~~~~~~
+
+The ``replaceOne()`` method requires the following parameters:
+
+- **Query filter** document, which determines which documents to replace.
+
+
+.. TODO: add link to read query filters 
+.. For more information about query filters, see :ref: <java-rs-query-filters>.
+
+- **Replace** document, which specifies the fields and values to insert in the new 
+  document.
+
+Replace Example
+---------------
+
+To replace a single document in a MongoDB collection, call the ``replaceOne()`` method
+and pass the query filter document as well as the replace document. Then, pass the ``replaceOne()`` result to the static
+``Mono.from()`` method from the ``Mono Publisher`` class. The ``Mono Publisher`` is a
+``Publisher`` implementation from the {+pr+} library. In {+java-rs+}, you must use
+``Publisher`` implementations to control how data is transmitted in your
+application. To learn more about the ``Mono`` class, see `Mono <https://projectreactor.io/docs/core/release/reference/#mono>`__ in the {+pr+} documentation.
+
+The following example uses the ``replaceOne()`` method to replace the fields and values of a
+document with a ``name`` field value of ``"Pizza Town"``. The ``replaceOne()``
+method replaces the original document with a document that has a ``name`` field
+value of ``"Mongo's Pizza"`` and a ``cuisine`` field value of ``"Pizza"``.
+
+.. code-block:: java
+   :copyable: true
+
+    Publisher<UpdateResult> replacePublisher = restaurants.replaceOne(
+            eq("name", "Pizza Town"),
+            new Document().append("name", "Mongo's Pizza")
+                    .append("cuisine", "Pizza"));
+    Mono.from(replacePublisher).block();
+
+Modify Replace Behavior
+-----------------------
+
+You can optionally modify the behavior of the ``replaceOne()`` method by calling
+an options method. The ``ReplaceOptions`` class provides methods that modify
+the behavior of the ``replaceOne()`` method. To use the ``ReplaceOptions``
+class, construct a new instance of the class, then call one or more of its methods
+to modify the replace operation. You can chain these method calls together.
+To modify the behavior of the replace operation, pass the class instance and
+chained method calls as the last argument to the
+``replaceOne()`` method.
+
+You can use the following methods in the ``ReplaceOptions`` class
+to modify the ``replaceOne()`` method.
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``bypassDocumentValidation(Boolean bypassDocumentValidation)``
```

Comment:
S: since the parameter label is just a label, can you shorten so that the table columns aren't so lopsided?
```suggestion
   * - ``bypassDocumentValidation(Boolean bypass)``
```